File: Easy/219. Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: list, k: int) -> bool:
        hashmap = {}
        for i in range(len(nums)):
            num = nums[i]
            if num in hashmap:
                if i - hashmap[num] <= k:
                    return True
                else:
                    hashmap[num] = i
            else:
                hashmap[num] = i
                    
        return False

File: Easy/219. Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int> map;
		for(int i=0; i<nums.size(); i++){
			int num = nums[i];
			if(map.count(num) > 0){
				if(i - map[num] <= k){
					return true;
				}else{
					map[num] = i;
				}
			}else{
				map[num] = i;
			}
		}
		return false;
    }
};